src/Coordination/KeeperDispatcher.cpp

```diff
@@ -83,30 +84,40 @@ void KeeperDispatcher::requestThread()
                 {
                     current_batch.emplace_back(request);
 
-                    /// Waiting until previous append will be successful, or batch is big enough
-                    /// has_result == false && get_result_code == OK means that our request still not processed.
-                    /// Sometimes NuRaft set errorcode without setting result, so we check both here.
-                    while (prev_result && (!prev_result->has_result() && prev_result->get_result_code() == nuraft::cmd_result_code::OK) && current_batch.size() <= max_batch_size)
+                    const auto try_get_request = [&]
```

Comment:
Probably, it'd be more natural to have `request` as in/out parameter in the lambda. Up to you.

Reply:
Yeah, but we don't use it instantly in the places where this function is called and if everything else is already captured don't see a reason to have `request` be different.